File: SourceCode/3.DomainServices/CodingTest.IDomainServices/Services/INewsLetterService.cs
using CodingTest.EntityModels;
using CodingTest.IDomainServices.Core;
using CodingTest.ServiceResponse;
using CodingTest.Utility.ParamViewModels;
using CodingTest.ViewModels;

namespace CodingTest.IDomainServices.Services
{
    public interface INewsLetterService : IBaseService<NewsLetterEntityModel, NewsLetterViewModel>
    {
        ResponseResults<NewsLetterViewModel> GetTopNewsLetters(long userId);
        ResponseResults<NewsLetterViewModel> GetNewsLetters(SearchNewLetterViewModel searchParam);
    }
}

File: SourceCode/1.UI/CodingTest.MVC.Angular.UI/MiniSpas/NewsLetterModule/Interfaces/INewsLetterService.ts
module NewsLetterModule.Interfaces
{
    export interface INewsLetterService
    {
        GetTopNewsLetters(userId:number): ng.IPromise<any>;
        GetNewsLetters( searchParam: NewsLetterModule.ViewModels.INewsLetterParamVM ): ng.IPromise<any>;
    }
}